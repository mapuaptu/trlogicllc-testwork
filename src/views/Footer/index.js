import Icon from '@/components/Icon/index.vue';

export default {
  name: 'Footer',
  components: {
    Icon,
  },
  props: {
    githubLink: {
      type: String,
      default: '',
    },
  },
};
