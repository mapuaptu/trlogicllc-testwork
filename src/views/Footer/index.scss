@import '@/styles/_variables.scss';

.footer {
  position: relative;
  display: flex;
  flex-flow: column;
  align-items: center;

  > *:not(:last-child) {
    margin-bottom: 5px;
  }

  a {
    display: flex;
    align-items: center;
    font-size: 12px;
    line-height: 12px;
    color: $color-java;
    transition: $transition;

    &:hover {
      opacity: 0.6;
    }
  }

  .icon {
    margin-right: 10px;
  }

  .copyright {
    font-size: 12px;
    line-height: 12px;
  }
}
