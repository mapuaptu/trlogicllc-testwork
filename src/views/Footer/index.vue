<template>
  <div :class="$style.footer">
    <a
      :href="githubLink"
      target="_blank"
    >
      <Icon
        icon="github"
        fill="#14b6bd"
        :class="$style.icon"
      />
      {{ githubLink }}
    </a>

    <div :class="$style.copyright">
      created by Alex Nikishkin
    </div>
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
