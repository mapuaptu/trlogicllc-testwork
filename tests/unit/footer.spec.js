import Footer from '@/views/Footer/index.vue';
import { mount } from '@vue/test-utils';

describe('Footer component', () => {
  it('1. Rendered success', async () => {
    const wrapper = await mount(Footer);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
