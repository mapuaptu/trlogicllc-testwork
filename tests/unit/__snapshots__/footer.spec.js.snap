// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component 1. Rendered success 1`] = `
<div><a href="" target="_blank">
    <div><svg width="24" height="24" viewbox="0 0 24 24">
        <path fill="#14b6bd"></path>
      </svg></div>

  </a>
  <div>
    created by Alex Nikishkin
  </div>
</div>
`;
